internal/internal_task_handlers.go

```diff
@@ -657,7 +663,10 @@ func (wth *workflowTaskHandlerImpl) getOrCreateWorkflowContext(
 	task *s.PollForDecisionTaskResponse,
 	historyIterator HistoryIterator,
 ) (workflowContext *workflowExecutionContextImpl, err error) {
-	metricsScope := wth.metricsScope.GetTaggedScope(tagWorkflowType, task.WorkflowType.GetName())
+	// We probably need to have the tasklist name in tags as well,
```

Comment:
Or not, combining with the implementation of scalable tasklist, it might result in 10x increase in the cardinality for that metric.